### DISCUSSION TITLE
What configuration changes are required from rabbitmq application side to enable fips support

### DISCUSSION THREAD

**Discussion by Vishnu-0809** [rabbitmq/rabbitmq-server#11876] [Questions] [answered]
Hi,
I am using erlang of 26.2.5 version and openssl-3.0.2, fips provider is enabled for openssl.
Now from the rabbitmq application pov what are the configuration changes do be done to enable fips mode.
I have tried {crypto, [{fips_mode, true}]} in advanced.config but the output of rabbitmqctl eval 'crypto:info_fips().' is still showing not_supported.
Please let me know what should be the config changes to enable fips for rabbitmq and what is the verification step to check that fips support is enabled from rabbitmq application.

**Comment by ansd** [comment]
VMware Tanzu RabbitMQ supports FIPS out of the box: https://tanzu.vmware.com/content/blog/introducing-vmware-tanzu-rabbitmq-3-13

**Reply by Vishnu-0809** [reply]
Thanks for the response, but I cannot use that because I am building rabbitmq from source and I should continue to do so

**Reply by michaelklishin** [reply, marked as answer]
@Vishnu-0809 if you build a custom RabbitMQ distribution, you are fully responsible for maintaining it.

Our team will not provide any support around FIPS-enabled Erlang. This feature is a part of the commercial edition that users looking for FIPS can buy.

**Comment by michaelklishin** [comment]
> Now from the rabbitmq application pov what are the configuration changes do be done to enable fips mode

Applications won't be aware of FIPS per se. FIPS restricts certain TLS features, so application must avoid using e.g. the [cipher suites](https://www.rabbitmq.com/docs/ssl#cipher-suites) that are not supported. In many cases this will be negotiated by the TLS handshake process transparently.
